src/Microsoft.Azure.WebJobs.Host/Listeners/FunctionListener.cs

```diff
@@ -54,7 +57,54 @@ public void Dispose()
             }
             catch (Exception e)
             {
-                new FunctionListenerException(_descriptor.ShortName, e).TryRecover(_trace, _logger);
+                var flx = new FunctionListenerException(_descriptor.ShortName, e);
+                if (_allowPartialHostStartup)
+                {
+                    flx.Handled = true;
+                }
+
+                flx.TryRecover(_trace, _logger);
+
+                if (_allowPartialHostStartup)
+                {
+                    // if to here, the listener exception was handled and
+                    // we're in partial startup mode.
+                    // we spin up a background task retrying to start the listener
+                    Task taskIgnore = Task.Run(() => RetryStartWithBackoffAsync(cancellationToken), cancellationToken);
+                }
+            }
+        }
+
+        public async Task RetryStartWithBackoffAsync(CancellationToken cancellationToken)
+        {
+            int attempt = 0;
+            while (!_started)
+            {
+                await Utility.DelayWithBackoffAsync(++attempt, cancellationToken, 
+                    unit: TimeSpan.FromSeconds(1),
+                    min: TimeSpan.FromSeconds(2),
+                    max: TimeSpan.FromMinutes(2));
+
+                try
+                {
+                    string message = $"Retrying to start listener for function '{_descriptor.ShortName}' (Attempt {attempt})";
+                    _trace.Info(message);
+                    _logger?.LogInformation(message);
+
```

Comment:
I wonder, is this going to be really noisy in the case of misconfigured functions where we never successfully start the listener? If so, can we use some sort of category here to make it easy to filter the noise out?

Reply:
What do you mean by "misconfigured"? Many issues will be indexing time failures (e.g. app setting for Connection doesn't exist, and in many cases even an invalid connection string for bindings like queue that do up front validation). If the function is truly broken, the backoff will max out to once every 2 minutes, which won't be very noisy. But yes, there will be an initial flurry of retries.

I'm open to putting them in a category